testing/posix/non_module_deps.bzl

```diff
@@ -0,0 +1,11 @@
+load(
+    "//tests:nixpkgs_repositories.bzl",
+    "nixpkgs_repositories",
+)
+
+def _non_module_deps_impl(ctx):
+    nixpkgs_repositories(bzlmod = True)
+
+non_module_deps = module_extension(
+    implementation = _non_module_deps_impl,
+)
```

Comment:
I think this is implied by your `TODO` comment in `testing/posix/MODULE.bazel`, but is the idea to later remove this `non_module_deps` extension and directly call the various `*_configure` macros that `nixpkgs_repositories` currently invokes once they've been modularised?

Reply:
Yes, exactly, that's the plan once https://github.com/tweag/rules_nixpkgs/issues/183 is implemented (or as part of it).